src/domainobjectfactories/back_office_position_factory.py

```diff
@@ -12,7 +12,7 @@ class BackOfficePositionFactory(Creatable):
     LEDGERS = ['TD', 'SD']
     PURPOSES = ['Outright', 'Obligation']
 
-    def create(self, record_count, start_id):
+    def create(self, record_count, start_id, lock=None):
```

Comment:
add lock parameter to all relevant docstrings - something like the below but adjust as appropriate depending on the method

```
lock : Lock
    Lock object from the multiprocessing module with global scope, shared among parallel processes 
    to define a critical section within the create method of the InstrumentFactory class - since 'create' is 
    an abstract method from the parent class Creatable, the lock is set to None by default for the other 
    factory subclasses
```